### PR TITLE
CY-1718: implementing manager status reporter

### PR DESCRIPTION
**Pull request by tyacbovi**
- implementing manager status reporter
- fixing missing timestamp in the status report
- fixing node id missing usage 
- fixing the token usage to work
- fixing cluster status endpoint for writing reports

### CODE REVIEW

File: status-reporter/status_reporter/status_reporter.py [hunk 1]
```diff
@@ -95,35 +98,43 @@ def __init__(self, sampler, node_type):
             raise InitializationError('Failed initialization of status '
                                       'reporter due to:\n {issues}'.
                                       format(issues='\n'.join(issues)))
+        self._node_id = self._config.get('node_id', None)
```
Comment (inbala) [inline status-reporter/status_reporter/status_reporter.py]: Don't we want to add the `_node_id` to the required check?

File: status-reporter/status_reporter/status_reporter.py [hunk 2]
```diff
@@ -95,35 +98,43 @@ def __init__(self, sampler, node_type):
...
+        self._reporter_credentials = {
+            'username': self._cloudify_user_name,
```
Comment (inbala) [inline status-reporter/status_reporter/status_reporter.py]: Do we need the user_name?

File: status-reporter/status_reporter/status_reporter.py (hunk 1 above)
Comment (tyacbovi) [reply status-reporter/status_reporter/status_reporter.py]: The field is written in the install process

File: status-reporter/status_reporter/status_reporter.py (hunk 2 above)
Comment (tyacbovi) [reply status-reporter/status_reporter/status_reporter.py]: it requires that

File: status-reporter/status_reporter/status_reporter.py (hunk 1 above)
Comment (inbala) [reply status-reporter/status_reporter/status_reporter.py]: Like managers_ips, cloudify_user_name and token, no?

Comment (tyacbovi) [reply status-reporter/status_reporter/status_reporter.py]: Well I will remove the 'cloudify_user_name' check after we will write that field automaticaly on install

File: rest-service/manager_rest/cluster_status_manager.py [hunk 3]
```diff
@@ -292,23 +275,36 @@ def get_cluster_status():
...
 
-def _verify_report_newer_than_current(report_time, status_path):
+def _are_keys_in_dict(dictionary, keys):
+    return all(key in dictionary for key in keys)
+
+
+def _verify_status_report_schema(node_id, report):
+    if (_are_keys_in_dict(report['report'], ['status', 'services'])
+        and report['report']['status'] in [ServiceStatus.HEALTHY,
+                                           ServiceStatus.FAIL]):
```
Comment (inbala) [inline rest-service/manager_rest/cluster_status_manager.py]: Wrong logic, it is valid if this 2 conditions are true

**Pull request merged**